frontend/javascripts/admin/task/task_search_form.tsx

```diff
@@ -50,279 +48,220 @@ const persistence = new Persistence<Pick<State, "fieldValues">>(
   "taskSearch",
 );
 
-class TaskSearchForm extends React.Component<Props, State> {
-  formRef = React.createRef<FormInstance>();
-  state: State = {
-    users: [],
-    projects: [],
-    taskTypes: [],
-    fieldValues: {},
-    isFetchingData: false,
-  };
+function TaskSearchForm({ onChange, initialFieldValues, isLoading, onDownloadAllTasks }: Props) {
+  const [form] = Form.useForm<TaskFormFieldValues>();
+  const [users, setUsers] = useState<APIUser[]>([]);
+  const [projects, setProjects] = useState<APIProject[]>([]);
+  const [taskTypes, setTaskTypes] = useState<APITaskType[]>([]);
+  const [fieldValues, setFieldValues] = useState<TaskFormFieldValues>({});
+  const [isFetchingData, setIsFetchingData] = useState(false);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
 
-  componentDidMount() {
-    this.fetchData();
+  useEffect(() => {
     // initialize form with default values when navigating from
     // project / taskType list views or when restoring values from persisted state
-    const persistedState = persistence.load();
-    const persistedFieldValues =
-      persistedState.fieldValues != null ? persistedState.fieldValues : {};
-    const fieldValues =
-      this.props.initialFieldValues != null ? this.props.initialFieldValues : persistedFieldValues;
+    const { fieldValues: persistedfieldValues } = persistence.load();
+    const persistedFieldValues = persistedfieldValues != null ? persistedfieldValues : {};
+    const fieldValues = initialFieldValues != null ? initialFieldValues : persistedFieldValues;
 
     if (_.size(fieldValues) > 0) {
-      const form = this.formRef.current;
-
-      if (!form) {
-        Toast.info(messages["ui.no_form_active"]);
-        return;
-      }
-
       form.setFieldsValue(fieldValues);
-      this.handleSearchFormFinish(false);
+      handleSearchFormFinish(false);
     }
-  }
+  }, [initialFieldValues]);
 
-  componentDidUpdate() {
-    persistence.persist(this.state);
-  }
+  useEffect(() => {
+    persistence.persist({ fieldValues: fieldValues });
+  }, [fieldValues]);
 
-  async fetchData() {
-    this.setState({
-      isFetchingData: true,
-    });
+  async function fetchData() {
+    setIsFetchingData(true);
     const [users, projects, taskTypes] = await Promise.all([
       getEditableUsers(),
       getProjects(),
       getTaskTypes(),
     ]);
-    this.setState({
-      users,
-      projects,
-      taskTypes,
-      isFetchingData: false,
-    });
+    setUsers(users);
+    setProjects(projects);
+    setTaskTypes(taskTypes);
+    setIsFetchingData(false);
   }
 
-  handleFormFinish = (
+  function handleFormFinish(
     isRandom: boolean,
     onFinishCallback: (arg0: QueryObject) => Promise<void>,
-    formValues: Record<string, any>,
-  ) => {
-    const queryObject: QueryObject = {};
-
-    if (formValues.taskId) {
-      const taskIds = formValues.taskId
-        .trim()
+    formValues: TaskFormFieldValues,
+  ) {
+    const queryObject: QueryObject = {
```

Comment:
Instead of checking and adding every value to the `QueryObject` individually, I find much more readable to assign them all in on go.

1. The type definition of `QueryObject` actually allows `undefined` values, so there is no need to "filter" them with a bunch of `if` statements.
2. If we want to filter them, something `_.omit(queryObject, _.isNil)` would filter them much shorter.